libs/utils/energy.py

```diff
@@ -325,4 +343,133 @@ def report(self, out_dir, out_energy='energy.json', out_samples='samples.csv'):
 
         return (channels_nrg, nrg_file)
 
+
+class IIOCapture(EnergyMeter):
+    "iio-capture Energy Meter for BayLibre ACME board"
+
+    def __init__(self, target, conf, res_dir):
+        super(IIOCapture, self).__init__(target)
+
+        self._iiocapturebin = conf['iiocapturebin']
+        self._hostname = conf['device_ip']
+        # Channels represent the connected probes
+        # iio:device<n>
+        self._channels = conf['iiodevice']
+        self._iio = [None] * len(self._channels)
+
+    def sample(self):
+        raise NotImplementedError('Not available for IIOCapture')
+
+    def reset(self):
+        """
+        Reset energy meter and start sampling from channels specified in the
+        target configuration.
+        """
+        for i, iiodevice in enumerate(self._channels):
+            # Terminate instance of iio-capture for this (channel, iiodevice)
+            if self._iio[i]:
+                self._iio[i].poll()
+                if self._iio[i].returncode is None:
+                    self._iio[i].terminate()
+                    self._iio[i].wait()
+                    self._iio[i] = None
+
+            csv_file = '{}/{}'.format(
+                self._res_dir,
+                'samples_{}.csv'.format(iiodevice)
+            )
+            self._iio[i] = Popen([self._iiocapturebin, "-n",
+                                  self._hostname, "-o",
+                                  "-c", "-f",
+                                  csv_file, iiodevice],
+                                 stdout=PIPE, stderr=STDOUT)
+
+            if self._iio[i] is None:
+                logging.error("%14s - Failed to run %s for %s", "IIOCapture",
+                              self._iiocapturebin, iiodevice)
+                logging.error("%14s - Make sure there are no iio-capture "
+                              "processes connected to %s and device %s",
+                              "IIOCapture", self._hostname, iiodevice)
+                continue
+
+            # Wait few milliseconds to check if there is any output.
+            # If this is the case, there must have been an error, so the user
+            # must be informed
+            sleep(0.02)
+            self._iio[i].poll()
+            if self._iio[i].returncode:
+                logging.error("%14s - Failed to run %s for %s", "IIOCapture",
```

Comment:
Just noticed that in the rest of this module we use single quotes for all logging messages... :-/